### PR TITLE
Output iteration results

### PR DESCRIPTION
**Pull request by chleh**
... after every iteration of the nonlinear solver.

This adds PR adds a callback to the nonlinear solvers' `solve()` method that can be used for output.
The output file name is the same as usual, but with `_iter_#ITER` appended.

The places in the code are not performance-critical, so it is runtime-checked if there shall be output. It can bve configured with the tag `<output_iteration_results>` in the output config.

### CODE REVIEW

File: ProcessLib/Output.h [hunk 1]
```diff
@@ -58,6 +58,15 @@ class Output
             const double t,
             GlobalVector const& x);
 
+    //! Writes output for the given \c process.
+    //! To be used for debug output after an iteration of the nonlinear solver.
+    void doOutputIteration(
```
Comment (norihiro-w) [inline ProcessLib/Output.h]: I would specify this is for nonlinear iterations, i.e. `doOutputNonlinearIteration`

File: ProcessLib/Output.h [hunk 2]
```diff
@@ -80,11 +89,13 @@ class Output
         MeshLib::IO::PVDFile pvd_file;
     };
 
-    Output(std::string const& prefix)
+    Output(std::string const& prefix, bool output_iteration_results)
```
Comment (norihiro-w) [inline ProcessLib/Output.h]: output_nonlinear_iteration_results

File: ProcessLib/Output.h (hunk 1 above)
Comment (chleh) [inline ProcessLib/Output.h]: Should the suffix then be `_nliter_#ITER`?

File: Applications/ApplicationsLib/UncoupledProcessesTimeLoop.h [hunk 3]
```diff
@@ -156,9 +156,9 @@ class UncoupledProcessesTimeLoop
 
     //! Solves one timestep for the given \c process.
     bool solveOneTimeStepOneProcess(
-            Vector& x, double const t, double const delta_t,
-            SingleProcessData& process_data,
-            Process& process);
+        Vector& x, std::size_t timestep, double const t, double const delta_t,
```
Comment (endJunction) [inline Applications/ApplicationsLib/UncoupledProcessesTimeLoop.h]: timestep could be const, isn't it?

File: ProcessLib/Output.cpp [hunk 4]
```diff
@@ -137,4 +142,27 @@ doOutputLastTimestep(Process const& process,
         doOutputAlways(process, timestep, t, x);
 }
 
+void Output::doOutputIteration(
+    Process const& process, unsigned timestep, const double t,
```
Comment (endJunction) [inline ProcessLib/Output.cpp]: timestep should be const

**Comment by endJunction** [comment]
After answering the questions :fast

**Comment by endJunction** [comment]
In the "[AppL] cleanup" a file Applications/ApplicationsLib/UncoupledProcessesTimeLoop.h.orig is removed. Please remove it from the repository. It is introduced in the first commit "optionally do output ..."

**Pull request merged**